locator/token_metadata.cc

```diff
@@ -570,8 +570,7 @@ std::unordered_map<inet_address, host_id> token_metadata_impl::get_endpoint_to_h
     std::unordered_map<inet_address, host_id> map;
     map.reserve(nodes.size());
     for (const auto& [endpoint, node] : nodes) {
-        // Restrict to members
-        if (!node->is_member()) {
+        if (node->left() || node->is_none()) {
```

Comment:
This change is not mentioned in the commit message. Does it fix some other bug?
Also, it probably belongs to a separate commit.

Reply:
Reverted it since we anyway don't display joining zero token nodes

Reply:
Added it back as per our discussion on other comment.

>Also, it probably belongs to a separate commit.

We are directly using this logic in scylla-nodetool.cc, so I think we need it here in this commit. Lemme know if otherwise

Reply:
You can split the PR into two commits, this change (which fixes and renames `get_endpoint_to_host_id_map`) will go to the first one, as it fixes mostly unrelated bug and it makes sense as a standalone change. And deserves its own commit message. The second commit would contain the rest of the changes

Reply:
Done